htdocs/xoops_lib/modules/protector/oninstall.php

```
@@ -11,11 +11,28 @@
+// used for table prefixes and template names inside the base function.
+if (!function_exists('xoops_module_install_protector')) {
+    /**
+     * XOOPS module lifecycle callback for protector install.
+     *
+     * @param XoopsModule $module
+     *
+     * @return bool
+     */
+    function xoops_module_install_protector($module)
+    {
+        $registry  = ProtectorRegistry::getInstance();
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'ProtectorRegistry' in method 'xoops_module_install_protector'. (undefined)

(StaticAccess)